src/Microsoft.Health.SqlServer/Features/Schema/SchemaInitializer.cs

```diff
@@ -232,6 +227,12 @@ private async Task<bool> CanInitializeAsync(CancellationToken cancellationToken)
                             return false;
                         }
                     }
+                    else
+                    {
+                        _logger.LogWarning("Database creation is not enabled.");
+
+                        throw new InvalidOperationException(Resources.NewDatabaseCannotBeCreated);
+                    }
```

Comment:
Before, if the database didn't exist and our config settings were set to the following:

`SqlServer:Initialize = true`
`SqlServer:AllowDatabaseCreation = false`

we would throw an error about incorrect login credentials and startup would fail. If there is no database and database creation is not enabled, it makes sense that startup would fail, but we should return a more useful error message. It wasn't that often that these two config settings were set to different values, so we never really ran into this.

Now that `SqlServer:Initialize` is removed and the error handling is updated, startup will fail if the database doesn't exist and `SqlServer:AllowDatabaseCreation = false`, and an appropriate message is returned.